src/main/webapp/scripts/script.js

```diff
@@ -1,4 +1,24 @@
 const API_KEY = '';
+const YOUTUBE_DISCOVERY = 'https://www.googleapis.com/discovery/v1/apis/youtube/v3/rest';
+
+function loadApi() {
+  return new Promise((resolve, reject) => {
+    gapi.load('client', function() {
```

Comment:
Nit: please convert all anonymous (unnamed) functions to arrow functions for consistency.

`gapi.load('client', () => {`